easypost/services/carrier_account_service.py

```diff
@@ -48,7 +52,11 @@ def retrieve(self, id: str) -> CarrierAccount:
 
     def update(self, id: str, **params) -> CarrierAccount:
         """Update a CarrierAccount."""
-        return self._update_resource(self._model_class, id, **params)
+        if params.get("type") in _UPS_OATH_CARRIER_ACCOUNT_TYPES:
```

Comment:
I am not sure if this will work. `Type` isn't a required field for update and we can't rely on it: https://docs.easypost.com/docs/carrier-accounts#update-a-carrieraccount

Reply:
Indeed. We'd need to retrieve it first then and pull out the `type` param. I don't love it but it'll make it easy on the users at the cost of a few more ms of time.